### PR TITLE
Merge master into null safety

### PR DESCRIPTION
**Pull request by iinozemtsev**
This brings the following changes from master branch:

- Support optional proto3 fields (#444)
- Fix handling of oneofs in dart proto3 json parsing. (#447)
- Use @dart = 2.7 in generated files (#455)
- Migrate from Travis to GitHub Actions (#460)
- Emit binary coded descriptors from dart protoc_plugin. (#406)

Github Actions config modified to run only on dev sdk. Also modify analyzer flags to remove `--fatal-infos`, there's one mixin-related violation:

```
hint • The library 'package:protoc_plugin/testing/mixins.dart' is legacy, and should not be imported into a null safe library. • out/protos/mixins.pb.dart:12:8 • import_of_legacy_library_into_null_safe
```

Also slightly speeding up a build by removing `make protos` step from `format` stage: generated protos are not checked in and formatted anyway, but generating and checking format takes 3-4 minutes.

### CODE REVIEW

File: protoc_plugin/lib/src/dart_options.pb.dart [hunk 1]
```diff
@@ -2,7 +2,7 @@
 //  Generated code. Do not modify.
 //  source: dart_options.proto
 //
-// @dart = 2.3
+// @dart = 2.7
```
Comment (sigurdm) [inline protoc_plugin/lib/src/dart_options.pb.dart:5]: Hmm - should this not be 2.12 for the null-safety branch?

**Review by sigurdm — APPROVED** [review]
LGTM - but perhaps you need to regenerate the descriptors...

**Comment by iinozemtsev** [comment]
No, see https://github.com/dart-lang/protobuf/pull/440, until protoc_plugin
is null safe we can't really migrate those. I'll send a separate PR to
change version on protoc sources to 2.11 and then regenerate bootstrap
protos, WDYT?

**Comment by sigurdm** [comment]
> until protoc_plugin is null safe we can't really migrate those.

Ah of course!

> I'll send a separate PR to
change version on protoc sources to 2.11 and then regenerate bootstrap
protos, WDYT?

Sounds good!

**Pull request merged**